cartographer_ros/cartographer_ros/map_builder_bridge.cc

```diff
@@ -246,11 +246,63 @@ MapBuilderBridge::GetTrajectoryStates() {
 
 visualization_msgs::MarkerArray MapBuilderBridge::GetTrajectoryNodeList() {
   visualization_msgs::MarkerArray trajectory_node_list;
+  // Compute the number of constrained nodes for each trajectory for
+  // inter-submap and inter-trajectory constraints.
+  std::map<int, int> trajectory_to_num_inter_submap_constrained_nodes;
+  std::map<int, int> trajectory_to_num_inter_trajectory_constrained_nodes;
+  const auto constraints = map_builder_->pose_graph()->constraints();
+  for (const auto& constraint : constraints) {
+    if (constraint.tag ==
+        cartographer::mapping::PoseGraph::Constraint::INTER_SUBMAP) {
+      if (constraint.node_id.trajectory_id ==
+          constraint.submap_id.trajectory_id) {
+        auto it = trajectory_to_num_inter_submap_constrained_nodes.find(
+            constraint.node_id.trajectory_id);
+        if (it != trajectory_to_num_inter_submap_constrained_nodes.end()) {
+          it->second = std::max(it->second, constraint.node_id.node_index);
```

Comment:
I think in some cases the latest constraint may also come from a (pretty new) submap and a (old) node, this should be added.

Reply:
Currently not supported by the public Cartographer API.